src/Compilers/VisualBasic/Test/Semantic/Binding/BindingObjectInitializerTests.vb

```diff
@@ -1119,9 +1119,8 @@ IObjectCreationExpression (Constructor: Sub C1..ctor()) (OperationKind.ObjectCre
           ISimpleAssignmentExpression (OperationKind.SimpleAssignmentExpression, Type: System.Func(Of System.Int32)) (Syntax: '.Field2 = F ... nd Function')
             Left: IFieldReferenceExpression: C1.Field2 As System.Func(Of System.Int32) (OperationKind.FieldReferenceExpression, Type: System.Func(Of System.Int32)) (Syntax: 'Field2')
                 Instance Receiver: IOperation:  (OperationKind.None) (Syntax: 'New C1 With ... d Function}')
-            Right: IConversionExpression (Implicit, TryCast: False, Unchecked) (OperationKind.ConversionExpression, Type: System.Func(Of System.Int32)) (Syntax: 'Function()  ... nd Function')
-                Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
-                Operand: IAnonymousFunctionExpression (Symbol: Function () As System.Int32) (OperationKind.AnonymousFunctionExpression, Type: null) (Syntax: 'Function()  ... nd Function')
+            Right: IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type: System.Func(Of System.Int32)) (Syntax: 'Function()  ... nd Function')
```

Comment:
>Right: IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type:  [](start = 12, length = 84)

Should this node be Implicit?

Reply:
I would think so. IsImplicit for VB conversions wasn't taking into account whether the cast was explicit in code. I've updated this code, and added a version of this test that binds just the Function() body to make sure it returns just the function body.

---
In reply to: [138739482](https://github.com/dotnet/roslyn/pull/22056#discussion_r138739482) [](ancestors = 138739482)